target/riscv/op_helper_cheri.c

```diff
@@ -49,24 +49,34 @@
 
 enum SCRAccessMode {
     SCR_Invalid = 0,
-    ASR_Flag = 1,
-    U_Always = (PRV_U + 1) << 1,
-    U_ASR = U_Always | ASR_Flag,
-    S_Always = (PRV_S + 1) << 1,
-    S_ASR = S_Always | ASR_Flag,
-    H_Always = (PRV_H + 1) << 1,
-    H_ASR = H_Always | ASR_Flag,
-    M_Always = (PRV_M + 1) << 1,
-    M_ASR = M_Always | ASR_Flag,
+    ASR_RW_Flag = 1,
+    ASR_W_Flag = 2,
```

Comment:
How about this:
```suggestion
    ASR_R_Flag = 1,
    ASR_W_Flag = 2,
    ASR_W_Flag = ASR_R_Flag | ASR_W_Flag,
```

Reply:
You mean ASR_RW_Flag =, surely?

Reply:
(if so, that's what I was imagining when discussing introducing ASR_RW_Flag)

Reply:
Yes exactly - I didn't see your suggestions until after I submitted these.